eta/core/datasets.py

```diff
@@ -2901,8 +2990,37 @@ def transform(self, src):
                 )
             )
 
+        if self.prepend_dataset_name:
+            for record in self._builder_dataset_to_merge.records:
+                base = os.path.basename(os.path.dirname(os.path.dirname(record.data_path)))
+                record.new_data_path = base + '_' + os.path.basename(record.data_path)
+                record.new_labels_path = base + '_' + os.path.basename(record.labels_path)
+
         src.add_container(self._builder_dataset_to_merge)
 
+class PrependDatasetNameToRecords(DatasetTransformer):
+    ''' Given a labeled dataset, this transformation prepends the dataset name
+    followed by an underscore to all data and label files in the dataset.
+    E.g. mydataset/data/vid.mp4 is now mydataset/data/mydataset_vid.mp4
+    '''
+
+    def transform(self, src):
+        '''Prepends the dataset name and an underscore to all records in the
+        dataset
+
+        Args:
+            src: a BuilderDataset
+        '''
+        for i in range(len(src.records)):
+            base = os.path.basename(os.path.dirname(os.path.dirname(
```

Comment:
I think this should be it's own private helper function `_get_dataset_name`. makes it clearer and then we can modify this and it will update everywhere if we ever need to

Reply:
for that matter, `prepend_to_name` could be a method on BuilderRecord